Sources/idt/idt.cc

```diff
@@ -126,16 +126,39 @@ struct PPCallbacks : clang::PPCallbacks {
   FileIncludes &file_includes_;
 };
 
+// Track a set of clang::Decl declarations by unique ID.
+class DeclSet {
+  std::set<std::uintptr_t> decls_;
+
+  // Use the raw address of the canonical declaration object as a unique
```

Comment:
I would phrase this as “use pointer identity” - this is a pretty common technique used in LLVM and ObjC.